crates/red_knot_python_semantic/src/types/call/bind.rs

```diff
@@ -746,6 +775,65 @@ impl<'db> Bindings<'db> {
                         }
                     }
 
+                    Some(KnownClass::TypeVar) => {
```

Comment:
Is it important to get all known cases into here rather than `infer.rs`, or does the combination of a) needing access to the containing assignment and b) needing to emit several diagnostics suggest that it might be simpler to just fully handle this case out in `infer.rs`?

Reply:
I had been trying to follow a general rule of "put as much here as possible", but I like your reasoning for moving it all into `infer.rs`.  That (combined with the semantic index suggestion) seems to have (mostly) solved the performance regression